app/Console/Commands/VideoData.php

```diff
@@ -77,34 +75,24 @@ public function handle()
 			return 0;
 		}
 
+		// Initialize factory for size variants
+		$sizeVariantFactory = resolve(SizeVariantFactory::class);
 		/** @var Photo $photo */
 		foreach ($photos as $photo) {
 			$this->line('Processing ' . $photo->title . '...');
-			$url = Storage::path('big/' . $photo->url);
-
-			if ($photo->thumbUrl != '') {
-				$thumb = Storage::path('thumb/') . $photo->thumbUrl;
-				if (file_exists($thumb)) {
-					$urlBase = explode('.', $photo->url);
-					$thumbBase = explode('.', $photo->thumbUrl);
-					if ($urlBase[0] !== $thumbBase[0]) {
-						$photo->thumbUrl = $urlBase[0] . '.' . $thumbBase[1];
-						rename($thumb, Storage::path('thumb/') . $photo->thumbUrl);
-						$this->line('Renamed thumb to match the video file');
-					}
-				}
-			}
+			$originalSizeVariant = $photo->size_variants->getSizeVariant(SizeVariant::ORIGINAL);
+			$fullPath = $originalSizeVariant->full_path;
 
-			if (file_exists($url)) {
-				$info = $this->metadataExtractor->extract($url, $photo->type);
+			if (file_exists($fullPath)) {
+				$info = $this->metadataExtractor->extract($fullPath, 'video');
 
 				$updated = false;
-				if ($photo->width == 0 && $info['width'] !== 0) {
-					$photo->width = $info['width'];
+				if ($originalSizeVariant->width == 0 && $info['width'] !== 0) {
+					$originalSizeVariant->width = $info['width'];
 					$updated = true;
```

Comment:
So would that `isDirty()` work here?

Reply:
You are asking whether `isDirty` is an alternative to using extra flag `$updated`? Yes, indeed it is. I fixed the code.